### PR TITLE
[Docs] Add upgrade notes for Next.js 9.5

### LINKED ISSUE #15719: [Docs] Add upgrade note to features released in Next.js 9.5

### PR DESCRIPTION
**Pull request by lfades**
Fixes https://github.com/vercel/next.js/issues/15719

### CODE REVIEW

File: docs/api-reference/next.config.js/headers.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ description: Add custom HTTP headers to your Next.js app.
 
 # Headers
 
+> This feature was introduced in [Next.js 9.5](https://nextjs.org/blog/next-9-5) and up. If you’re using older versions of Next.js, please upgrade before trying it out.
```
Comment (ijjk) [inline docs/api-reference/next.config.js/headers.md:7]: Should we mention this for `basePath` and `trailingSlash` also?

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| buildDuration | 9.5s | 9.7s | ⚠️ +214ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.722 | 1.683 | -0.04 |
| / avg req/sec | 1451.43 | 1485.2 | +33.77 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 0.929 | 0.927 | 0 |
| /error-in-render avg req/sec | 2692.07 | 2697.86 | +5.79 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-935e413..31c4.js gzip | 6.76 kB | 6.76 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 56.8 kB | 56.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-4b8f4a8..dule.js gzip | 5.84 kB | 5.84 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 51.8 kB | 51.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| Overall change | 30.8 kB | 30.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-6f8445b..99e1.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.72 kB | 7.72 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-91516ae..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _buildManifest.js gzip | 274 B | 274 B | ✓ |
| _buildManife..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| index.html gzip | 945 B | 945 B | ✓ |
| link.html gzip | 953 B | 953 B | ✓ |
| withRouter.html gzip | 940 B | 940 B | ✓ |
| Overall change | 2.84 kB | 2.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| buildDuration | 10.3s | 10.6s | ⚠️ +285ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-935e413..31c4.js gzip | 6.76 kB | 6.76 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 56.8 kB | 56.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-4b8f4a8..dule.js gzip | 5.84 kB | 5.84 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 51.8 kB | 51.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| Overall change | 30.8 kB | 30.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-6f8445b..99e1.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.72 kB | 7.72 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-91516ae..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _buildManifest.js gzip | 274 B | 274 B | ✓ |
| _buildManife..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _error.js | 1.02 MB | 1.02 MB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.82 kB | 3.82 kB | ✓ |
| index.js | 1.02 MB | 1.02 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.2 MB | 5.2 MB | ✓ |

</details>

</details>
Commit: 6e324cdebdd6eeb62b1271be66239c399bba0485

File: docs/api-reference/next.config.js/headers.md (hunk 1 above)
Comment (lfades) [reply docs/api-reference/next.config.js/headers.md:7]: Yes! 🐛

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| buildDuration | 14.5s | 14.7s | ⚠️ +150ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.728 | 2.718 | -0.01 |
| / avg req/sec | 916.32 | 919.72 | +3.4 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.642 | 1.637 | 0 |
| /error-in-render avg req/sec | 1522.53 | 1526.79 | +4.26 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-935e413..31c4.js gzip | 6.76 kB | 6.76 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 56.8 kB | 56.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-4b8f4a8..dule.js gzip | 5.84 kB | 5.84 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 51.8 kB | 51.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| Overall change | 30.8 kB | 30.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-6f8445b..99e1.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.72 kB | 7.72 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-91516ae..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _buildManifest.js gzip | 274 B | 274 B | ✓ |
| _buildManife..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| index.html gzip | 945 B | 945 B | ✓ |
| link.html gzip | 953 B | 953 B | ✓ |
| withRouter.html gzip | 940 B | 940 B | ✓ |
| Overall change | 2.84 kB | 2.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| buildDuration | 16s | 16.2s | ⚠️ +197ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-935e413..31c4.js gzip | 6.76 kB | 6.76 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 56.8 kB | 56.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-4b8f4a8..dule.js gzip | 5.84 kB | 5.84 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 51.8 kB | 51.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| Overall change | 30.8 kB | 30.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-6f8445b..99e1.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.72 kB | 7.72 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-91516ae..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _buildManifest.js gzip | 274 B | 274 B | ✓ |
| _buildManife..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _error.js | 1.02 MB | 1.02 MB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.82 kB | 3.82 kB | ✓ |
| index.js | 1.02 MB | 1.02 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.2 MB | 5.2 MB | ✓ |

</details>

</details>
Commit: 31445974bda8000e460029746f1a51a2191273e2

**Review by Timer — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| buildDuration | 11.9s | 12.3s | ⚠️ +379ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.048 | 2.162 | ⚠️ +0.11 |
| / avg req/sec | 1220.73 | 1156.56 | ⚠️ -64.17 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.116 | 1.187 | ⚠️ +0.07 |
| /error-in-render avg req/sec | 2240.27 | 2106.51 | ⚠️ -133.76 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-935e413..31c4.js gzip | 6.76 kB | 6.76 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 56.8 kB | 56.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-4b8f4a8..dule.js gzip | 5.84 kB | 5.84 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 51.8 kB | 51.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| Overall change | 30.8 kB | 30.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-6f8445b..99e1.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.72 kB | 7.72 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-91516ae..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _buildManifest.js gzip | 274 B | 274 B | ✓ |
| _buildManife..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| index.html gzip | 945 B | 945 B | ✓ |
| link.html gzip | 953 B | 953 B | ✓ |
| withRouter.html gzip | 940 B | 940 B | ✓ |
| Overall change | 2.84 kB | 2.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| buildDuration | 12.9s | 13.6s | ⚠️ +708ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-935e413..31c4.js gzip | 6.76 kB | 6.76 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 56.8 kB | 56.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-4b8f4a8..dule.js gzip | 5.84 kB | 5.84 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 51.8 kB | 51.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| Overall change | 30.8 kB | 30.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-6f8445b..99e1.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.72 kB | 7.72 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-91516ae..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _buildManifest.js gzip | 274 B | 274 B | ✓ |
| _buildManife..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | lfades/next.js add-custom-routes-note | Change |
| - | - | - | - |
| _error.js | 1.02 MB | 1.02 MB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.82 kB | 3.82 kB | ✓ |
| index.js | 1.02 MB | 1.02 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.2 MB | 5.2 MB | ✓ |

</details>

</details>
Commit: 585fdf71306ea038ff4b024a00e86baee6d13a14

**Pull request merged**